src/TogglePassword/src/Form/TogglePasswordTypeExtension.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\UX\TogglePassword\Form;
+
+use Symfony\Component\Form\AbstractTypeExtension;
+use Symfony\Component\Form\Extension\Core\Type\PasswordType;
+use Symfony\Component\Form\FormInterface;
+use Symfony\Component\Form\FormView;
+use Symfony\Component\OptionsResolver\Options;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+use Symfony\Component\Translation\TranslatableMessage;
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+/**
+ * @author Félix Eymonot <felix.eymonot@alximy.io>
+ */
+final class TogglePasswordTypeExtension extends AbstractTypeExtension
+{
+    public function __construct(private readonly ?TranslatorInterface $translator)
+    {
+    }
+
+    public static function getExtendedTypes(): iterable
+    {
+        return [PasswordType::class];
+    }
+
+    public function configureOptions(OptionsResolver $resolver): void
+    {
+        $resolver->setDefaults([
+            'toggle' => false,
+            'hidden_label' => 'Hide',
+            'visible_label' => 'Show',
+            'hidden_icon' => 'Default',
+            'visible_icon' => 'Default',
+            'button_classes' => ['toggle-password-button'],
+            'toggle_container_classes' => ['toggle-password-container'],
+            'toggle_translation_domain' => null,
```

Comment:
I'm legitimately not sure, but do we need to make this configurable? I think it's more common to hardcode one: you're grouping your translations into a group, then telling the user which group/domain to use when translating.

If we used something like `ux_toggle_password`, we would then provide a `ux_toggle_password.en.xlf` file. It would simply translate "Show" to "Show".

Reply:
The approach was to allow users to implement their own custom labels with custom translations, to avoid implementing X translations files inside the bundle (for only 2 words each). Allowing custom translation domain would allow user to define or follow their own translation domain implementation, and it sorts of follow the same approach as label_translation_domain in others `FormType` objects